spectral_cube/dask_spectral_cube.py

```diff
@@ -0,0 +1,794 @@
+        ----------
+        threshold : float
+            The ``sigma`` parameter in `astropy.stats.sigma_clip`, which refers
+            to the number of sigma above which to cut.
+        verbose : int
+            Verbosity level to pass to joblib
+
+        """
+
+        def spectral_sigma_clip(array):
+            result = stats.sigma_clip(array, sigma=threshold, axis=0, **kwargs)
+            return result.filled(np.nan)
+
+        # Rechunk so that there is only one chunk spectrally and let dask
+        # decide for the rest
+        return self._map_blocks_to_cube(spectral_sigma_clip,
+                                        rechunk=(-1, 'auto', 'auto'))
+
+    def spectral_smooth(self,
+                        kernel,
+                        convolve=convolution.convolve,
+                        verbose=0,
+                        num_cores=None,
+                        **kwargs):
+        """
+        Smooth the cube along the spectral dimension
+
+        Note that the mask is left unchanged in this operation.
+
+        Parameters
+        ----------
+        kernel : `~astropy.convolution.Kernel1D`
+            A 1D kernel from astropy
+        convolve : function
+            The astropy convolution function to use, either
+            `astropy.convolution.convolve` or
+            `astropy.convolution.convolve_fft`
+        verbose : int
+            Verbosity level to pass to joblib
+        kwargs : dict
+            Passed to the convolve function
+        """
+
+        if isinstance(kernel.array, u.Quantity):
+            raise u.UnitsError("The convolution kernel should be defined "
+                               "without a unit.")
+
+        def spectral_smooth(array):
+            if array.size > 0:
+                kernel_3d = kernel.array.reshape((len(kernel.array), 1, 1))
+                return convolve(array, kernel_3d, normalize_kernel=True)
+            else:
+                return array
+
+        # Rechunk so that there is only one chunk spectrally
+        return self._map_blocks_to_cube(spectral_smooth,
+                                        rechunk=(-1, 'auto', 'auto'))
+
+    def spectral_smooth_median(self, ksize,
+                               use_memmap=True,
+                               verbose=0,
+                               num_cores=None,
+                               **kwargs):
+        """
+        Smooth the cube along the spectral dimension
+
+        Parameters
+        ----------
+        ksize : int
+            Size of the median filter (scipy.ndimage.filters.median_filter)
+        verbose : int
+            Verbosity level to pass to joblib
+        kwargs : dict
+            Not used at the moment.
+        """
+
+        if not SCIPY_INSTALLED:
+            raise ImportError("Scipy could not be imported: this function won't work.")
+
+        def median_filter_wrapper(img, **kwargs):
+            return ndimage.median_filter(img, (ksize, 1, 1), **kwargs)
```

Comment:
If `ksize` is not an `int`, this fails with a `TypeError`. A type conversion or check with an helpful error message would be useful.  Somewhere inputting a `float` in the `SpectralCube` version is corrected for, though I don't see where...

Reply:
I think it's because here we pass a tuple of values not a scalar, which maybe causes issues.